src/app/brands/[brandId]/page.tsx

```diff
@@ -0,0 +1,84 @@
+import BrandPage from "@/components/BrandPage/BrandPage";
+import ProductGallery from "@/components/ProductGallery";
+import { getBrand } from "@/model/BrandModel";
+import { getFavorites } from "@/model/FavoriteProductsModel";
+import { getProducts } from "@/model/ProductModel";
+import { Product } from "@/services/ProductAPI";
+import getUserServerSide from "@/services/server/UserService";
+import { Metadata } from "next";
+import { notFound } from "next/navigation";
+
+export async function generateMetadata({
+  params,
+}: {
+  params: { brandId: string };
+}): Promise<Metadata> {
+  let brand = await getBrand(params.brandId);
+
+  if (brand === undefined) {
+    return {
+      title: "Not found!",
+    };
+  }
+  return {
+    title: brand.name,
+  };
+}
+
+export default async function Page({
+  params,
+}: {
+  params: { brandId: string };
+}) {
+  const user = await getUserServerSide();
+  let products: Product[] = [];
+  let brand = undefined;
+
+  try {
+    [products, brand] = await Promise.all([
+      getProducts({
+        brandId: params.brandId,
+      }),
+      getBrand(params.brandId),
+    ]);
+  } catch (err) {
+    console.error(err);
+  }
+  let favoriteProductsIds: string[] = [];
+
+  if (brand === undefined) {
+    return notFound();
+  }
+
+  if (products.length === 0) {
+    return (
+      <BrandPage user={user}>
+        <p className="font-bold text-center text-xl"> {brand.name}</p>
+        <p className="m-2 text-center">There are no products for this brand</p>
+      </BrandPage>
+    );
+  }
+
+  if (user !== undefined) {
+    try {
+      let favorites = await getFavorites(user.id);
+      favoriteProductsIds = favorites.map((favorite) => favorite.id);
+    } catch (err) {
+      console.error("Failed to show favorite products", err);
```

Comment:
Technically we should show an error message here as well. Try and think of a way in which we don't duplicate a lot of code, but if we fail here we show a `<p>` below the one with the brand name

```
 <p className="font-bold text-center text-xl"> {brand.name}</p>
```